src/xcode/ENA/ENA/Source/Scenes/Events/Store/EventCheckoutService.swift

```diff
@@ -0,0 +1,176 @@
+////
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import UserNotifications
+
+final class EventCheckoutService {
+
+	// MARK: - Init
+
+	init(
+		eventStore: EventStoringProviding,
+		contactDiaryStore: DiaryStoringProviding,
+		userNotificationCenter: UserNotificationCenter = UNUserNotificationCenter.current()
+	) {
+		self.eventStore = eventStore
+		self.contactDiaryStore = contactDiaryStore
+		self.userNotificationCenter = userNotificationCenter
+
+		registerToDidBecomeActiveNotification()
```

Comment:
maybe we need a deinit with unregister that notification here as well?

Reply:
Good point! Checked it and I learned today, that its not needed since iOS 9.0:
"If your app targets iOS 9.0 and later or macOS 10.11 and later, and you used addObserver(_:selector:name:object:), you do not need to unregister the observer. If you forget or are unable to remove the observer, the system cleans up the next time it would have posted to it."